File: client/MauiChat/Models/GroupDto.cs
namespace MauiChat.Models;

/// <summary>Matches server <c>GroupDto</c>: id, name, avatarUrl, memberCount, createdAt.</summary>
public class GroupDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? AvatarUrl { get; set; }
    public int MemberCount { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: client/flutter_chat/lib/models/group_dto.dart
class GroupDto {
  final String id;
  final String name;
  final String? avatarUrl;
  final int memberCount;
  final DateTime createdAt;

  const GroupDto({
    required this.id,
    required this.name,
    this.avatarUrl,
    required this.memberCount,
    required this.createdAt,
  });

  factory GroupDto.fromJson(Map<String, dynamic> j) => GroupDto(
        id: j['id'] as String,
        name: (j['name'] as String?) ?? '',
        avatarUrl: j['avatarUrl'] as String?,
        memberCount: (j['memberCount'] as int?) ?? 0,
        createdAt: DateTime.parse(j['createdAt'] as String),
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'avatarUrl': avatarUrl,
        'memberCount': memberCount,
        'createdAt': createdAt.toIso8601String(),
      };
}
